File: Week 1/credit.c
#include <stdio.h>
#include <cs50.h>


int two(long numb) {
    int other_two = numb % 10;
    other_two = other_two * 2;
    int temp10 = other_two / 10;
    other_two = other_two % 10 + temp10;
    return other_two;
}

int one(long numb) {
    int other_one = numb % 10;
    return other_one;
}

long snip(long v) {
    v = v - (v % 10);
            v = v/10;
            return v;
}



//Main function

int main(void) {

    long visa = 0;
    bool check = true;

    do {
    visa = get_long("Please enter a credit card number\n");
    if (visa >= 0 && visa <= 9999999999999999) {

        check = false;}

    } while (check == true);

    long visa_u = visa;
    int sum = 0;
    int digits = 0;

    //Check for digits and change digits variable

    if (visa >= 1000000000000000 && visa <= 9999999999999999) {
        digits = 16;
    }
    else if (visa >= 1000000000000 && visa <= 9999999999999)  {
        digits = 13;
    }
    else if (visa >= 100000000000000 && visa <= 999999999999999) {
        digits = 15;
    }
    else {
        digits = 1;
    }

    //Luhn's algorithm

    for (int i = 1; i <= digits; i++) {
        if (i % 2 == 1) {
            sum = sum + one(visa);
            visa = snip(visa);
        }
        else if (i % 2 == 0) {
            sum = sum + two(visa);
            visa = snip(visa);
        }
    }

    if (sum % 10 == 0) {
        if (digits == 15 && (visa_u / 10000000000000 == 34 || visa_u / 10000000000000 == 37)) {
            printf("AMEX\n");
        }
        else if (digits == 13) {
            printf("VISA\n");
        }
        else if ((digits == 16) && ((visa_u / 100000000000000) >= 51 && (visa_u / 100000000000000) <= 55)) {
            printf("MASTERCARD\n");
        }
        else if (digits == 16 && visa_u / 1000000000000000 == 4) {
            printf("VISA\n");
        }
        else {
            int debug = visa_u / 100000000000000;
            printf("INVALID\n");
        }
    }
    else {
        printf("INVALID\n");
    }




}

File: Week 6/Python/credit.py
from cs50 import get_int

def visa_check(number):
    length = len(str(number))
    print(length)
    """for i in str(number):
        if type(i) != int:
            return False"""
    if length != 15 or length != 16 or length != 13:
        return False
    else:
        return True

#Get input
visa = get_int("Please enter a credit card number.\n")


length = len(str(visa))
visa_array = [int(i) for i in str(visa)]


#Luhn's algorithm

def luhn(number):
    #1. Create array with all digits
    digits_array =  []
    digits_array = [int(i) for i in str(number)]

    #2. Multiply every other digit by two
    for i in range(len(digits_array)):
        if length % 2 == 1:
            if i % 2 == 1:
                #Odd
                digits_array[i] *= 2
                #Add up the two digits together
                if digits_array[i] >= 10:
                    oneplace = digits_array[i] % 10
                    tenplace = digits_array[i] // 10
                    digits_array[i] = oneplace + tenplace
            if i % 2 == 0:
                #Even
                pass
        elif length % 2 == 0:
            if i % 2 == 0:
                #Even
                digits_array[i] *= 2
                #Add up the two digits together
                if digits_array[i] >= 10:
                    oneplace = digits_array[i] % 10
                    tenplace = digits_array[i] // 10
                    digits_array[i] = oneplace + tenplace
            if i % 2 == 1:
                #Odd
                pass

    #4. Sum of all values
    total = sum(digits_array)
    #print(total)
    #5. Return final digit
    final_digit = total % 10

    #print(final_digit)

    if final_digit == 0:
        return True
        #It is a credit card
    if final_digit != 0:
        return False
        #Not legit credit card


if luhn(visa) == False:
    print("INVALID\n")
elif luhn(visa) == True:
    if length == 15:
        print("AMEX\n")
    elif length == 16 and visa_array[0] == 5:
        print("MASTERCARD\n")
    elif length == 13 or length == 16:
        if visa_array[0] == 4:
            print("VISA\n")




